ivy/functional/frontends/tensorflow/math.py

```diff
@@ -419,6 +419,21 @@ def unsorted_segment_sqrt_n(
     return x
 
 
+@to_ivy_arrays_and_back
+def unsorted_segment_prod(
+    data, segment_ids, num_segments, name="unsorted_segment_prod"
+):
+    ivy.utils.assertions.check_equal(list(segment_ids.shape), [list(data.shape)[0]])
+    x = ivy.ones(tuple([num_segments] + list(data.shape[1:])))
+    count = ivy.ones((num_segments,))
+    for i in range(segment_ids.shape[0]):
+        x[segment_ids[i]] = x[segment_ids[i]] * data[i]
+        count[segment_ids[i]] += 1
+    for j in range(num_segments):
+        x[j] = ivy.pow(x[j], 1.0 / count[j])
+    return x
```

Comment:
Also, I'm not sure why you are using `count`. It seems like you are averaging, but I think the function is supposed to calculate just the product.